src/utils/gauges.ts

```diff
@@ -198,19 +191,32 @@ export async function getGaugeData(
       gaugeNamesPromise,
       gaugeKillStatusesPromise,
     ])
+
     const minterSDLRate = await (gaugeMinterContract
       ? gaugeMinterContract.rate()
       : Promise.resolve(Zero))
 
+    const lpTokenToPool: Partial<{
+      [lpToken: string]: {
+        poolName: string
+        poolAddress: string
+      }
+    }> = Object.values(basicPools || {}).reduce(
+      (prevData, { lpToken, poolAddress, poolName }) => {
+        return {
+          ...prevData,
+          [lpToken]: { poolAddress, poolName },
+        }
+      },
+      {},
+    )
+
     const gauges: LPTokenAddressToGauge = gaugeAddresses.reduce(
       (previousGaugeData, gaugeAddress, index) => {
         const lpTokenAddress = gaugeLpTokenAddresses[index]?.toLowerCase()
-        const poolAddress = gaugePoolAddresses[index]
+        const pool = lpTokenToPool[lpTokenAddress]
         const isValidPoolAddress = Boolean(
-          poolAddress && !isAddressZero(poolAddress),
-        )
-        const gaugePool = Object.values(basicPools || {}).find(
-          (pool) => pool.poolAddress === poolAddress,
+          pool && pool.poolAddress && !isAddressZero(pool.poolAddress),
```

Comment:
you should be able to shorten this to `pool?.poolAddress && ...`

Reply:
agreed, I don't think poolAddress would ever be 0x0 here

Reply:
Furthermore since you just created that map and you know that each key has a valid value, you can just say `const pool = lpTokenToPool[lpTokenAddress] as MyPoolType` so you don't have to do the null checks later on

Reply:
@ug02fast, I'm running into the issue of having to check for both `undefined` as well as `null` types. This is mainly coming from our need to use `?` optional chaining, but the `.tryAll` method is returning `Promise<null>` if the call fails.

What do you think about checking for both, or we pick 1 type of falsy value and go with that?